airflow/models/baseoperator.py

```diff
@@ -636,51 +636,50 @@ def __setstate__(self, state):
         self.__dict__ = state
         self._log = logging.getLogger("airflow.task.operators")
 
-    def render_template_from_field(self, attr, content, context, jinja_env):
+    def render_template(
+        self, content: Any, context: Dict, jinja_env: Optional[jinja2.Environment] = None
+    ) -> Any:
         """
-        Renders a template from a field. If the field is a string, it will
-        simply render the string and return the result. If it is a collection or
-        nested set of collections, it will traverse the structure and render
-        all elements in it. If the field has another type, it will return it as it is.
+        Render a templated object. The object may be nested and will be templated recursively.
+
+        :param content: Content to template. Only strings can be templated (may be nested).
```

Comment:
I'm not quite sure what the "may be nested" means given we've said only strings can be accepted, but also the code accepts more than just strings.

BTW: the type comments don't currently show up in the docs, so it's probably worth adding `:type context:` etc :(

Reply:
Indeed only strings will ultimately be templated. The message to convey is that you can also supply e.g. a list of strings, which will be templated. A list itself won't be templated, but the function will iterate over all elements in a list and template if type == str, hence the "may be nested".

Does that make sense and should I word it differently?

Reply:
I'll add the types in the docs, and will check out https://pypi.org/project/sphinx-autodoc-typehints/ separate from this PR.

Reply:
I didn't get that from the docs, so some of what you said there could do with being added to the doc comment please